arch/arm/core/aarch64/cache.c

```diff
@@ -0,0 +1,53 @@
+/* cache.c - d-cache support for AARCH64 CPUs */
+
+/*
+ * Copyright 2020 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief d-cache manipulation
+ *
+ * This module contains functions for manipulation of the d-cache.
+ */
+
+#include <kernel.h>
+#include <cache.h>
+
+#define	CTR_EL0_DMINLINE_SHIFT	16
+#define	CTR_EL0_DMINLINE_MASK	0xF
+#define	CTR_EL0_CWG_SHIFT	24
+#define	CTR_EL0_CWG_MASK	0xF
+
+int arch_dcache_flush(void *addr, size_t size);
+int arch_dcache_invd(void *addr, size_t size);
+
+int arch_dcache_range(void *addr, size_t size, int op)
+{
+	if (op == K_CACHE_INVD) {
+		arch_dcache_invd(addr, size);
+	} else if (op == K_CACHE_WB_INVD) {
+		arch_dcache_flush(addr, size);
+	} else {
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+size_t arch_cache_line_size_get(void)
+{
+	uint32_t ctr_el0;
```

Comment:
`uint64_t`

Reply:
No. ctr_el0 is 32bits register.

Reply:
are you sure? In the DDI0487E it looks like it is a 64bit register (even though the relevant bits are in the first 32 bits).

Reply:
DDI0487A shows it is 32bit. I'll use uint64_t according the new version.